server/stats.go

```diff
@@ -176,6 +257,38 @@ func triplesToStatsVar(
 	return &statsVar, nil
 }
 
+// getValue get the stat value from ObsTimeSeries.
+// When date is given, it get the value from the highest ranked source series
+// that has the date.
+// When date is not given, it get the latest value from the highest ranked
+// source series.
+func getValue(in *ObsTimeSeries, date string) (float64, error) {
+	if in == nil {
+		return 0, fmt.Errorf("Nil obs time series for getLatest()")
+	}
+	sourceSeries := in.SourceSeries
+	sort.Sort(byRank(sourceSeries))
+	if date != "" {
+		for _, series := range sourceSeries {
+			if value, ok := series.Val[date]; ok {
+				return value, nil
+			}
+		}
+		return 0, fmt.Errorf("No data found for date %s", date)
+	}
```

Comment:
By sorting first and then checking date, we miss out if the 2nd best series has the date we want right?

Goes back to the above comment on whether the ranking logic should take date into account.

Reply:
After sorting, this checks the highest ranking series first, if found the date then return it, otherwise will check the second best series and go on this. Basically if multiple series have the same date then this picks from the highest ranked series.

Reply:
Ah, I misread. This LG